lib/provider/data_types.go

```diff
@@ -105,3 +105,14 @@ type Snapshot struct {
 	// tags for the snapshot
 	SnapshotTags SnapshotTags `json:"tags"`
 }
+
+// AuthorizationRequest capture details of autorization to be made
+type AuthorizationRequest struct {
+	// Volume to update the authorization
+	Volume Volume
```

Comment:
Can we have only ID instead of Volume struct as only volume ID is required.

Reply:
We want to be as much generic as  possible in defining the interfaces , Not sure which providers wants which all parameters.